backend/graph/api/schema.resolvers.go

```diff
@@ -173,36 +174,64 @@ func (r *mutationRootResolver) SendSupportEmail(ctx context.Context, title strin
 }
 
 // CompleteTask is the resolver for the completeTask field.
-func (r *mutationRootResolver) CompleteTask(ctx context.Context, id string) (bool, error) {
+func (r *mutationRootResolver) CompleteTask(ctx context.Context, id string, selectedAlternatives []string) (bool, error) {
 	p, err := getProfile(ctx)
 	if err != nil {
 		return false, err
 	}
+
 	task, err := getTask(ctx, r.Resolver, id)
 	if err != nil {
 		return false, err
 	}
-	storedIds, err := r.Loaders.CompletedTasksLoader.Get(ctx, p.ID)
-	if err != nil {
-		return false, err
+
+	if task.ComptetitionMode {
+		storedIds, err := r.Loaders.CompletedTasksLoader.Get(ctx, p.ID)
+		if err != nil {
+			return false, err
+		}
+
+		if lo.Contains(utils.PointerArrayToArray(storedIds), task.ID) {
+			return false, common.ErrTaskAlreadyCompleted
+		}
 	}
```

Comment:
Note: this is not entirely correct. There should be a "locked" flag that signifies the user has submitted the final answer.